platform/source/tools/clustercli/common.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2015-2017 Applatix, Inc. All rights reserved.
+#
+
+import re
+
+from anytree import Node
+from .cli import AbstractPrompt
+
+
+class CommonPrompts(AbstractPrompt):
+
+    INSTALLCMD = "install"
+    UNINSTALLCMD = "uninstall"
+
+    def __init__(self):
+        self.aws_profiles = self._get_profiles()
+        self._profile_names = [x for x in self.aws_profiles]
+
+        self.root = Node("name",
+                    prompt=u'Enter a cluster name',
+                    validator=u'^[a-zA-Z0-9-_]{2,20}$',
+                    help=u'This is a friendly name for your cluster. Minimum 2 characters, Maximum 20. Only a-z, A-Z, 0-9, - and _ allowed',
+                    default=u'argo-cluster'
+                    )
+
+        profileNode = Node("profile",
+                           prompt=u'Enter the AWS profile to use',
+                           values=self._profile_names,
+                           default=u'default' if "default" in self._profile_names else self._profile_names[0],
+                           help=u'Select an AWS profile to install the cluster',
+                           parent=self.root
+                           )
+
+    def get_root(self):
+        return self.root
+
+    @staticmethod
+    def _get_profiles():
+        from boto3.session import Session
+        s = Session()
+        return s._session.full_config["profiles"]
```

Comment:
I would be hesitate to access a protected member of a class in open-sourced code. But since boto3 is not providing a method to access all profiles and it's not a good idea to write our own yaml parser as well, we really need to make a decision between code style and usability.

Personally I'd not do this but it's your call.

Reply:
Unless these is a better way in boto, I see no reasonable alternative.